src/main/java/com/simprints/biometrics/simface/detection/MlKitFaceDetectionProcessor.kt

```diff
@@ -85,54 +87,59 @@ internal class MlKitFaceDetectionProcessor(
 
     private fun calculateFaceQuality(
         face: Face,
-        imageWidth: Int,
-        imageHeight: Int,
+        image: Bitmap,
     ): Float {
         return try {
-            var score = 0.0
-
-            // These should add to 1.0
-            val faceRotationWeight = 0.3
-            val faceTiltWeight = 0.05
-            val faceNodWeight = 0.05
-            val faceSizeWeight = 0.3
-            val eyeOpennessWeight = 0.3
-
-            // Face Rotation Score
-            score += faceRotationWeight * (1.0 - (face.headEulerAngleY.absoluteValue / 90.0))
-
-            // Face Tilt Score
-            score += faceTiltWeight * (1.0 - (face.headEulerAngleZ.absoluteValue / 90.0))
-
-            // Face Nod Score
-            score += faceNodWeight * (1.0 - (face.headEulerAngleX.absoluteValue / 90.0))
-
-            // Face Size Relative to Image Size
-            val faceArea = face.boundingBox.width() * face.boundingBox.height()
-            val imageArea = imageWidth * imageHeight
-            score += faceSizeWeight * (faceArea.toDouble() / imageArea)
-
-            // Eye Openness Score
-            score += eyeOpennessWeight * calculateEyeOpennessScore(face)
-
-            // TODO: Blur Detection
-            // TODO: Brightness and Contrast Score
-
-            // Just in case limit to 0-1 range
-            return score.coerceIn(0.0, 1.0).toFloat()
+            val boundingBox = face.boundingBox.clampToBounds(image.width, image.height)
+            val faceBitmap = Bitmap.createBitmap(
+                image,
+                boundingBox.left,
+                boundingBox.top,
+                boundingBox.width(),
+                boundingBox.height()
+            )
+
+            val parameters = QualityParameters(
+                maxAlignmentAngle = 20.0,
+                maxIndividualAngle = 25.0,
+                minBlur = 50.0,
+                maxBlur = 100.0,
+                minBrightness = 30.0,
+                optimalBrightnessLow = 80.0,
+                optimalBrightnessHigh = 150.0,
+                maxBrightness = 190.0,
+                brightnessSteepness = 0.3,
+                minContrast = 30.0,
+                maxContrast = 47.0
+            )
+
+            val weights = QualityWeights(
+                alignment = 0.28,
+                blur = 0.3,
+                brightness = 0.1,
+                contrast = 0.3,
+                eyeOpenness = 0.02
+            )
+
+            val qualityScore = SimQ.calculateFaceQuality(
```

Comment:
SimQ instance should be provided in the constructor for at least for testing/mocking purposes, but also for better control of resource initialisation.